src/appmixer/kafka/jobs.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+const connections = require('./connections');
+
+module.exports = async (context) => {
+
+    const config = require('./config')(context);
+
+    await context.scheduleJob('reconnect', config.reconnectJob.schedule, async () => {
+
+        const registeredComponents = await context.service.loadState();
+
+        for (const component of registeredComponents) {
+            await context.log('info', 'Reconnecting the component.');
+            await connections.addConnection(context, component.value);
+        }
+    });
+
+    await context.scheduleJob('disconnect', config.disconnectJob.schedule, async () => {
```

Comment:
But there is still the problem with lots of connections. Even with concurrency, it can run for let's say a minute. You `loadState` and get 1000 connections, going through them and opening them. In the middle, one (many) flow is stopped, state is changed, but the job does not know about it and will open the connection. Yes, it won't do anything and if a message arrives, it will be stopped. But it would be good, not to open it in the first place.

Solution. Before actually opening/closing a connection from the job, you have to check again, if it still is in the `state`.